packages/test/test-utils/src/testContainerRuntimeFactory.ts

```diff
@@ -83,12 +84,21 @@ export const createTestContainerRuntimeFactory = (
 		}
 
 		public async instantiateFirstTime(runtime: ContainerRuntime): Promise<void> {
-			const rootContext = runtime.createDetachedRootDataStore([this.type], "default");
+			// Back-compat - old code does not return IDataStore for rootContext.attachRuntime() call!
+			// Thus need to leverage old API createDetachedRootDataStore() that is gone in latest releases.
+			const rootContext =
+				"createDetachedRootDataStore" in runtime
+					? (runtime as any).createDetachedRootDataStore([this.type], "default")
+					: runtime.createDetachedDataStore([this.type], "default");
```

Comment:
We don't need to create detached data store here right. We can create a regular aliased data store at least in new (non back-compat scenarios).

Reply:
Great suggestion! For some reason I assume that aliasing feature was not in 1.3.7 (and thus I'd still deal with legacy), but it does not look like the case (for  createDataStore()), so yes, I can use it!

Reply:
Unfortunately, this is not possible - see https://github.com/microsoft/FluidFramework/pull/19951.